pkg/clustermanager/cluster_manager.go

```diff
@@ -505,6 +506,11 @@ func (c *ClusterManager) EKSAClusterSpecChanged(ctx context.Context, cluster *ty
 		return true, nil
 	}
 
+	if !newClusterSpec.OIDCConfig.Spec.Equal(&currentClusterSpec.OIDCConfig.Spec) {
```

Comment:
maybe move this before the provider check? can potentially run less check. Also the current `default` statement logic is not right. This method will already return `true` if it's a non vsphere provider.

Reply:
The default check for the vsphere data center will be addressed in a different pr after adding a case for docker.